ebib-utils.el

```diff
@@ -2376,25 +2392,19 @@ year can be extracted from DATE, return nil."
   (ebib-clean-TeX-markup (ebib-get-field-value field key db "" 'unbraced 'xref)))
 
 (defun ebib-clean-TeX-markup (string)
-  "Return STRING without TeX markup."
+  "Return STRING without TeX markup.
+
+Any substring matching the car of a cell in
+`ebib-TeX-markup-replace-alist' is replaced with the
+correspnoding cdr, if the cdr is a string, or with the result of
+calling the cdr on the match (if it is a function). This is done
+with `replace-regexp-in-string', which see for details."
   (save-match-data
-    (let ((case-fold-search t))
-      ;; First replace TeX commands with their arguments and do some
-      ;; transformations, i.e., \textsc{cls} ==> CLS.  Note that optional
-      ;; arguments are also removed.
-      (while (string-match "\\\\\\([a-zA-Z*]*\\)\\(?:\\[.*?\\]\\)*{\\(.*?\\)}" string)
-        (let ((arg (copy-sequence (match-string 2 string))))
-          (pcase (match-string 1 string)
-            ((or "emph" "textit") (setq arg (propertize arg 'face '(italic))))
-            ("textbf" (setq arg (propertize arg 'face '(bold))))
-            ("textsc" (setq arg (upcase arg))))
-          (setq string (replace-match arg t t string)))))
-    ;; Replace as defined in `ebib-TeX-markup-replace-alist', and
-    ;; remove all {braces} (this takes care of nested braces too)
-    (replace-regexp-in-string
-     (rx-to-string (append '(or) (mapcar #'car ebib-TeX-markup-replace-alist)))
-     (lambda (match) (cdr (assoc match ebib-TeX-markup-replace-alist 'string=)))
-     string)))
+    (cl-loop for (ptn . rep) in ebib-TeX-markup-replace-alist
```

Comment:
I'd prefer `pattern` and `replacement` here as variable names, TBH.

Reply:
Can do.